cluster-operator/src/main/java/io/strimzi/operator/cluster/model/VolumeUtils.java

```diff
@@ -405,4 +409,62 @@ private static String getVolumeNameHashStub(String name)   {
+                    // skipping if a volume with same Secret name was already added
+                    if (!volumeList.stream().anyMatch(v -> v.getName().equals(volumeName))) {
+                        volumeList.add(VolumeUtils.createSecretVolume(volumeName, volumeName, isOpenShift));
+                    }
+                }
+            }
+        }
+
+        AuthenticationUtils.configureClientAuthenticationVolumes(authentication, volumeList, oauthVolumeNamePrefix, isOpenShift, volumeNamePrefix, createOAuthSecretVolumes);
+    }
+
+    protected static void getVolumeMounts(KafkaClientAuthentication authentication, ClientTls tls, List<VolumeMount> volumeList, String tlsCertBasedVolumeMount, String passwordVolumeMount, String oauthTlsCertsBaseVolumeMount, Rack rack) {
+        getVolumeMounts(authentication, tls, volumeList, tlsCertBasedVolumeMount, passwordVolumeMount, oauthTlsCertsBaseVolumeMount, "oauth-certs", "", false, null, null);
+    }
+
+    protected static void getVolumeMounts(KafkaClientAuthentication authentication, ClientTls tls, List<VolumeMount> volumeList, String tlsCertBasedVolumeMount, String passwordVolumeMount, String oauthTlsCertsBaseVolumeMount, String oauthVolumeNamePrefix, Rack rack) {
+        getVolumeMounts(authentication, tls, volumeList, tlsCertBasedVolumeMount, passwordVolumeMount, oauthTlsCertsBaseVolumeMount, oauthVolumeNamePrefix, "", false, null, null);
+    }
+
+    protected static List<VolumeMount> getVolumeMounts(KafkaClientAuthentication authentication, ClientTls tls, List<VolumeMount> volumeMountList, String tlsCertBasedVolumeMount, String passwordVolumeMount, String oauthTlsCertsBaseVolumeMount, String oauthVolumeNamePrefix, String volumeNamePrefix, boolean createOAuthSecretVolumes, String oauthVolumeMountPath, String alias) {
+
+        if (tls != null) {
+            List<CertSecretSource> trustedCertificates = tls.getTrustedCertificates();
+
+            if (trustedCertificates != null && trustedCertificates.size() > 0) {
+                for (CertSecretSource certSecretSource : trustedCertificates) {
+                    String volumeMountName = alias != null ? alias + '-' + certSecretSource.getSecretName() : certSecretSource.getSecretName();
+                    // skipping if a volume mount with same Secret name was already added
+                    if (!volumeMountList.stream().anyMatch(vm -> vm.getName().equals(volumeMountName))) {
+                        volumeMountList.add(VolumeUtils.createVolumeMount(volumeMountName,
+                                tlsCertBasedVolumeMount + certSecretSource.getSecretName()));
+                    }
+                }
+            }
+        }
+
+        AuthenticationUtils.configureClientAuthenticationVolumeMounts(authentication, volumeMountList, tlsCertBasedVolumeMount, passwordVolumeMount, oauthTlsCertsBaseVolumeMount, oauthVolumeNamePrefix, volumeNamePrefix, createOAuthSecretVolumes, oauthVolumeMountPath);
+
+        return volumeMountList;
+    }
```

Comment:
You should think about the naming. These methods are fairly specific - you cannto use them in brokers for example. So maybe they should have some more descriptive name than just `getVolumes` and `getVolumeMounts`. Adding some Javadoc might help as well.